### DISCUSSION TITLE
python docker images updates based on pyproject.toml restrictions

### DISCUSSION THREAD

**Discussion by rhuanbarreto** [renovatebot/renovate#14513] [Suggest an Idea]
Today the updates proposed in dockerfiles doesn't count on restrictions in pyproject.toml. Is that possible to have such restriction updated together?

**Comment by rarkins** [comment]
I don't understand the question. Can you provide and example?

**Reply by rhuanbarreto** [reply]
Let's say you have a python project (poetry managed) that runs on a docker container. You have these two files:

`pyproject.toml`
```
...
[tool.poetry.dependencies]
python = ">=3.9,<3.10"
...
```

`Dockerfile`
```
FROM --platform=linux/amd64 python:3.9-slim as build
...
```

Renovate comes with a PR trying to update python in the Dockerfile to 3.10, but in pyproject.toml 3.10 is not compatible. So either both should be updated together or no update PR would be created.

**Reply by rarkins** [reply]
The answer is "updated together". Please create a reproduction repository to demonstrate and we'll take a look

**Reply by rhuanbarreto** [reply]
There you go https://github.com/rhuanbarreto/renovate-test/pull/2

**Reply by rarkins** [reply]
Can you copy/paste the `packageFiles with updates` debug log message for that repo?

**Reply by rhuanbarreto** [reply]
```json
DEBUG: packageFiles with updates
{
  "config": {
    "dockerfile": [
      {
        "packageFile": "Dockerfile",
        "deps": [
          {
            "depName": "python",
            "currentValue": "3.9-slim",
            "replaceString": "python:3.9-slim",
            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "datasource": "docker",
            "depType": "final",
            "depIndex": 0,
            "warnings": [],
            "versioning": "docker",
            "currentVersion": "3.9",
            "isSingleVersion": true,
            "fixedVersion": "3.9-slim",
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "3.10",
                "newValue": "3.10-slim",
                "newMajor": 3,
                "newMinor": 10,
                "updateType": "minor",
                "branchName": "renovate/python-3.x"
              }
            ]
          }
        ]
      }
    ],
    "poetry": [
      {
        "packageFile": "pyproject.toml",
        "deps": [
          {
            "depName": "Flask",
            "depType": "dependencies",
            "currentValue": "^2.0.3",
            "managerData": {
              "nestedVersion": false
            },
            "datasource": "pypi",
            "versioning": "poetry",
            "depIndex": 0,
            "warnings": [],
            "homepage": "https://palletsprojects.com/p/flask",
            "changelogUrl": "https://flask.palletsprojects.com/changes/",
            "currentVersion": "2.0.3",
            "updates": []
          }
        ],
        "registryUrls": null,
        "constraints": {
          "python": ">=3.9,<3.10"
        },
        "lockFiles": [
          "poetry.lock"
        ]
      }
    ]
  }
}
```

**Reply by rarkins** [reply]
So per that, python is not detected as a *dependency* within the poetry project to begin with. It seems we just put it into `constraints.python`. Therefore, the next step is a feature request to treat `python` as a dependency from here: https://github.com/rhuanbarreto/renovate-test/blob/4c528e45cfe9c5f4fec78d7bcf0ef4e7527cbcf2/pyproject.toml#L8-L9

I'm not sure why we ignore it today.

**Reply by rhuanbarreto** [reply]
Done. #14649

**Reply by simon-liebehenschel** [reply]
> Done. #14649

@rhuanbarreto There is no link in your message. Was this bug fixed? Could you please clarify? Is there an existing issue to fix the problem?

I use `renovate/renovate:34.28` and Renovate updates Python version in all my `Dockerfile`s to the latest (3.11) version. At the same time, Renovate does not update Python version inside a "pyproject.toml" (I have 3.10).

**Reply by rhuanbarreto** [reply]
Issue is still on. But I couldn't be more clear to the renovate team so they could potentially fix. I don't know why, but they deleted my issue in the end.

**Reply by rarkins** [reply]
That may have been an accident. Normally issues show up as deleted after they're converted to discussions

**Reply by rhuanbarreto** [reply]
The issue is still actual. My [PR](https://github.com/rhuanbarreto/renovate-test/pull/2) is still only updating the dockerfile independently from the `pyproject.toml` file.
